stubs/pygit2/pygit2/__init__.pyi

```diff
@@ -0,0 +1,423 @@
+    GIT_REPOSITORY_INIT_BARE as GIT_REPOSITORY_INIT_BARE,
+    GIT_REPOSITORY_INIT_EXTERNAL_TEMPLATE as GIT_REPOSITORY_INIT_EXTERNAL_TEMPLATE,
+    GIT_REPOSITORY_INIT_MKDIR as GIT_REPOSITORY_INIT_MKDIR,
+    GIT_REPOSITORY_INIT_MKPATH as GIT_REPOSITORY_INIT_MKPATH,
+    GIT_REPOSITORY_INIT_NO_DOTGIT_DIR as GIT_REPOSITORY_INIT_NO_DOTGIT_DIR,
+    GIT_REPOSITORY_INIT_NO_REINIT as GIT_REPOSITORY_INIT_NO_REINIT,
+    GIT_REPOSITORY_INIT_RELATIVE_GITLINK as GIT_REPOSITORY_INIT_RELATIVE_GITLINK,
+    GIT_REPOSITORY_INIT_SHARED_ALL as GIT_REPOSITORY_INIT_SHARED_ALL,
+    GIT_REPOSITORY_INIT_SHARED_GROUP as GIT_REPOSITORY_INIT_SHARED_GROUP,
+    GIT_REPOSITORY_INIT_SHARED_UMASK as GIT_REPOSITORY_INIT_SHARED_UMASK,
+    GIT_REPOSITORY_OPEN_BARE as GIT_REPOSITORY_OPEN_BARE,
+    GIT_REPOSITORY_OPEN_CROSS_FS as GIT_REPOSITORY_OPEN_CROSS_FS,
+    GIT_REPOSITORY_OPEN_FROM_ENV as GIT_REPOSITORY_OPEN_FROM_ENV,
+    GIT_REPOSITORY_OPEN_NO_DOTGIT as GIT_REPOSITORY_OPEN_NO_DOTGIT,
+    GIT_REPOSITORY_OPEN_NO_SEARCH as GIT_REPOSITORY_OPEN_NO_SEARCH,
+    GIT_REPOSITORY_STATE_APPLY_MAILBOX as GIT_REPOSITORY_STATE_APPLY_MAILBOX,
+    GIT_REPOSITORY_STATE_APPLY_MAILBOX_OR_REBASE as GIT_REPOSITORY_STATE_APPLY_MAILBOX_OR_REBASE,
+    GIT_REPOSITORY_STATE_BISECT as GIT_REPOSITORY_STATE_BISECT,
+    GIT_REPOSITORY_STATE_CHERRYPICK as GIT_REPOSITORY_STATE_CHERRYPICK,
+    GIT_REPOSITORY_STATE_CHERRYPICK_SEQUENCE as GIT_REPOSITORY_STATE_CHERRYPICK_SEQUENCE,
+    GIT_REPOSITORY_STATE_MERGE as GIT_REPOSITORY_STATE_MERGE,
+    GIT_REPOSITORY_STATE_NONE as GIT_REPOSITORY_STATE_NONE,
+    GIT_REPOSITORY_STATE_REBASE as GIT_REPOSITORY_STATE_REBASE,
+    GIT_REPOSITORY_STATE_REBASE_INTERACTIVE as GIT_REPOSITORY_STATE_REBASE_INTERACTIVE,
+    GIT_REPOSITORY_STATE_REBASE_MERGE as GIT_REPOSITORY_STATE_REBASE_MERGE,
+    GIT_REPOSITORY_STATE_REVERT as GIT_REPOSITORY_STATE_REVERT,
+    GIT_REPOSITORY_STATE_REVERT_SEQUENCE as GIT_REPOSITORY_STATE_REVERT_SEQUENCE,
+    GIT_STASH_APPLY_PROGRESS_ANALYZE_INDEX as GIT_STASH_APPLY_PROGRESS_ANALYZE_INDEX,
+    GIT_STASH_APPLY_PROGRESS_ANALYZE_MODIFIED as GIT_STASH_APPLY_PROGRESS_ANALYZE_MODIFIED,
+    GIT_STASH_APPLY_PROGRESS_ANALYZE_UNTRACKED as GIT_STASH_APPLY_PROGRESS_ANALYZE_UNTRACKED,
+    GIT_STASH_APPLY_PROGRESS_CHECKOUT_MODIFIED as GIT_STASH_APPLY_PROGRESS_CHECKOUT_MODIFIED,
+    GIT_STASH_APPLY_PROGRESS_CHECKOUT_UNTRACKED as GIT_STASH_APPLY_PROGRESS_CHECKOUT_UNTRACKED,
+    GIT_STASH_APPLY_PROGRESS_DONE as GIT_STASH_APPLY_PROGRESS_DONE,
+    GIT_STASH_APPLY_PROGRESS_LOADING_STASH as GIT_STASH_APPLY_PROGRESS_LOADING_STASH,
+    GIT_STASH_APPLY_PROGRESS_NONE as GIT_STASH_APPLY_PROGRESS_NONE,
+)
+from .packbuilder import PackBuilder as PackBuilder
+from .remotes import Remote as Remote
+from .repository import Repository
+from .settings import Settings
+from .submodules import Submodule as Submodule
+
+features: enums.Feature
+LIBGIT2_VER: tuple[int, int, int]
+
+def init_repository(
+    path: str | bytes | PathLike[AnyStr_co] | None,
```

Comment:
A TypeVar doesn't make sense here since there is only one occurrence in the function. In any case, covariance is meaningless in a function.

You can instead use the `StrOrBytesPath` from `_typeshed`.

Reply:
Thanks for the suggestion; I only recently returned to Python development so I'm still getting familiar with the new typing scheme. I'll push an updated version later.